Sources/Components/SelectionBoxes/Selectionbox.swift

```diff
@@ -0,0 +1,178 @@
+            delegate?.selectionbox(self, didUnselectItem: removedItem)
+        }
+    }
+}
+
+/* Base class for selections */
+
+public protocol SelectionboxDelegate: class {
+    func selectionbox(_ selectionbox: Selectionbox, didSelectItem item: SelectionboxItem)
+    func selectionbox(_ selectionbox: Selectionbox, didUnselectItem item: SelectionboxItem)
+}
+
+public class Selectionbox: UIView {
+    public var unselectedImage: UIImage? {
+        didSet {
+            for item in stack.arrangedSubviews as! [SelectionboxItem] {
+                item.imageView.image = unselectedImage
+            }
+        }
+    }
+
+    public var unselectedAnimationImages: [UIImage]? {
+        didSet {
+            for item in stack.arrangedSubviews as! [SelectionboxItem] {
+                item.imageView.animationImages = unselectedAnimationImages
+                if let unselected = unselectedAnimationImages {
+                    item.imageView.unselectedDuration = Double(unselected.count) / 60.0
+                }
+            }
+        }
+    }
+
+    public var selectedImage: UIImage? {
+        didSet {
+            for item in stack.arrangedSubviews as! [SelectionboxItem] {
+                item.imageView.highlightedImage = selectedImage
+            }
+        }
+    }
+
+    public var selectedAnimationImages: [UIImage]? {
+        didSet {
+            for item in stack.arrangedSubviews as! [SelectionboxItem] {
+                item.imageView.highlightedAnimationImages = selectedAnimationImages
+                if let selected = selectedAnimationImages {
+                    item.imageView.selectedDuration = Double(selected.count) / 60.0
+                }
+            }
+        }
+    }
+
+    public var title: String? {
+        get { return titleLabel.text }
+        set { titleLabel.text = newValue }
+    }
+
+    public weak var delegate: SelectionboxDelegate?
+
+    // MARK: Private properties
+
+    private var highlightedItem: SelectionboxItem?
+
+    private let titleLabel: UILabel = {
+        let label = Label(style: .title3)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    private let stack: UIStackView = {
+        let stack = UIStackView(frame: .zero)
+        stack.axis = .vertical
+        stack.distribution = .fillEqually
+        stack.translatesAutoresizingMaskIntoConstraints = false
+        return stack
+    }()
+
+    // MARK: Implementation
+
+    public init(strings: [String]) {
+        super.init(frame: .zero)
+        setupBoxes(with: strings)
+    }
+
+    public override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        guard let touch = touches.first else { return }
+        let location = touch.location(in: self)
+        guard let item = hitTest(location, with: event) as? SelectionboxItem else { return }
+        highlightedItem = item
+    }
+
+    public override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
```

Comment:
I really like how you handle touch detection, works quite well 👏

Reply:
Thanks! :)